test/controllers/api/v2/activation_keys_controller_test.rb

```diff
@@ -80,12 +80,23 @@ def test_show_protected
       end
     end
 
+    def stub_view_environment(env)
+      Katello::KTEnvironment.stubs(:readable).returns(stub(:find_by => env))
+    end
+
+    def stub_view_content_view(view)
+      Katello::ContentView.stubs(:readable).returns(stub(:find_by => view))
+    end
+
     def test_create_protected
+      stub_view_environment(@library)
+      stub_view_content_view(@view)
+
```

Comment:
You could do something similar to https://github.com/Katello/katello/pull/8957/files#diff-950d343e8ded0e9a59fbe8561c825bb6L427-L440 instead of stubbing. I dont mind either way